core/domain/user_jobs_one_off.py

```diff
@@ -254,3 +255,39 @@ def map(item):
     @staticmethod
     def reduce(key, stringified_values):
         pass
+
+
+class UserLastExplorationActivityOneOffJob(jobs.BaseMapReduceJobManager):
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [user_models.UserSettingsModel]
+
+    @staticmethod
+    def map(user_model):
+        user_id = user_model.id
+        contributions = user_models.UserContributionsModel.get(user_id)
+
+        created_explorations = exp_services.get_multiple_explorations_by_id(
+            contributions.created_exploration_ids)
+        if created_explorations:
+            user_model.last_created_an_exploration = max(
+                [model.created_on for model in created_explorations.values()])
+
+        user_commits = (
+            exp_models.ExplorationCommitLogEntryModel.query(
+                exp_models.ExplorationCommitLogEntryModel.user_id == user_id).
+            order(-exp_models.ExplorationCommitLogEntryModel.created_on).
+            fetch())
+
+        for commit in user_commits:
+            if (commit.exploration_id not in created_explorations or
```

Comment:
Oh, I was making sure that created exploration should not be considered for checking edit time but I think 2nd condition of checking commit type is enough for that. We won't need this one.
I will remove it. Now whether 2nd condition should be kept or not depends on Rahim's response. If edit time is only for edited explorations (exclude created exps) then we need 2nd condition to make sure that given commit_type is not 'create'. If it also includes created exps then we dont need any condition. 

To answer your first question (what this conditions do), in first condition I am trying to remove all commits of exploration that user has created (created explorations is a dict with exploration_id as key and domain object as value). But this can result in wrong time, since a user can also edit his own exploration later on and that commit has to be considered for checking last edit time. That's why I added 2nd condition (to remove all create commit_type) and after adding 2nd condition 1st is not necessary (I realized now).

Thanks